24th-App-Team-1-iOS/Feature/MessageFeature/Sources/Presentation/Components/MessageToggleView.swift

```diff
@@ -59,17 +70,37 @@ final class MessageToggleView: UIView {
     }
     
     private func setupAutoLayout() {
-        
-        homeButton.snp.makeConstraints {
+        mainButton.snp.makeConstraints {
             $0.verticalEdges.equalToSuperview()
-            $0.width.equalToSuperview()
+            $0.left.equalToSuperview().inset(20)
+            $0.width.equalTo(self).multipliedBy(0.5).offset(-20)
+        }
+        resultButton.snp.makeConstraints {
+            $0.verticalEdges.equalToSuperview()
+            $0.right.equalToSuperview().inset(20)
+            $0.width.equalTo(self).multipliedBy(0.5).offset(-20)
         }
-        
         
         underLine.snp.makeConstraints {
             $0.height.equalTo(1)
             $0.horizontalEdges.equalToSuperview()
             $0.bottom.equalToSuperview().offset(-1)
         }
+        
+        selectedLine.snp.makeConstraints {
+            $0.left.equalToSuperview().offset(20)
+            $0.height.equalTo(2)
+            $0.width.equalTo(self).multipliedBy(0.5).offset(-20)
+            $0.bottom.equalToSuperview().offset(-2)
+        }
+    }
+    
+    private func updateToggleLayout(_ isSelected: Bool) {
+        UIView.animate(withDuration: 0.3, delay: 0, options: .curveEaseInOut) { [weak self] in
```

Comment:
```suggestion
        UIView.animate(withDuration: 0.3, delay: 0, options: .curveEaseInOut) {
```
- `UIView.animate` 는 타입 메서드로 구성되어 있어서 [weak self]를 사용하지 않아도 순한 참조 발생을 일으키지 않아용 ☺️